src/v2/cookbook/avoiding-memory-leaks.md

```diff
@@ -0,0 +1,139 @@
+
+## Simple Example
+
+The following example shows a memory leak caused by using the Choices.js library in a Vue component and not properly cleaning it up. Later, we will show how to remove the Choices.js footprint and avoid the memory leak.
+
+In the example below, we load up a select with a lot of options and then we use a show/hide button with a `v-if` directive to add it and remove it from the virtual DOM. The problem with this example is that the v-if directive removes the parent element from the DOM, but we did not clean up the additional DOM pieces created by Choices.js, causing a memory leak. 
+
+```html
+<link rel='stylesheet prefetch' href='https://joshuajohnson.co.uk/Choices/assets/styles/css/choices.min.css?version=3.0.3'>
+<script src='https://joshuajohnson.co.uk/Choices/assets/scripts/dist/choices.min.js?version=3.0.3'></script>
+
+<div id="app">
+  <button v-if="showChoices" @click="hide">Hide</button>
+  <button v-if="!showChoices" @click="show">Show</button>
+  <div v-if="showChoices">
+    <select id="choices-single-default"></select>
+  </div>
+</div>
+```
+```js
+const app = new Vue({
+  el: "#app",
+  data: function() {
+    return {
+      showChoices: true
+    }
+  },
+  mounted: function() {
+    this.initializeChoices()
+  },
+  methods: {
+    initializeChoices() {
+      let list = []
+      // lets load up our select with many choices 
+      // so it will use a lot of memory
+      for (let i = 0; i < 1000; i++) {
+        list.push({
+          label: "Item " + i,
+          value: i
+        })
+      }
+      new Choices("#choices-single-default", {
+        searchEnabled: true,
+        removeItemButton: true,
+        choices: list
+      })
+    },
+    show() {
+      this.showChoices = true
+      this.$nextTick(() => {
+        this.initializeChoices()
+      })
+    },
+    hide() {
+      this.showChoices = false
+    }
+  }
+})
+```
+To see this memory leak in action, open this [CodePen example](https://codepen.io/freeman-g/pen/qobpxo) using Chrome and then open the Chrome Task Manager (Chrome Menu > More Tools > Task Manager). Now, click the show/hide button 50 or so times. You should see the memory usage in the Chrome Task Manager increase and never be reclaimed.
+
+![Memory Leak Example](/images/memory-leak-example.png)
+
+## Resolving the Memory Leak
+
+In the above example, we can use our `hide()` method to do some clean up and solve the memory leak prior to removing the select from the DOM. To accomplish this, we will keep a property in our Vue instance’s data object and we will use the [Choices API’s](https://github.com/jshjohnson/Choices) `destroy()` method to perform the clean up.
+
+Check the memory usage again with this [updated CodePen example](https://codepen.io/freeman-g/pen/mxWMor).
+
+```js
+const app = new Vue({
+  el: "#app",
+  data: function() {
```

Comment:
In the vue docs, we separate function with a space, like this `function ()`. I had an issue with this because of Prettier, so if that's your case too and you use VS Code, you can use this extension to toggle it on and off for the file: https://marketplace.visualstudio.com/items?itemName=tombonnike.vscode-status-bar-format-toggle